modules/apps/product-navigation/product-navigation-product-menu-api/src/test/java/com/liferay/product/navigation/product/menu/display/context/ProductMenuDisplayContextTest.java

```diff
@@ -0,0 +1,231 @@
+import com.liferay.product.navigation.applications.menu.configuration.ApplicationsMenuInstanceConfiguration;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.ArgumentMatchers;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Anderson Luiz
+ */
+public class ProductMenuDisplayContextTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() throws Exception {
+		_themeDisplay = Mockito.mock(ThemeDisplay.class);
+
+		_mockHttpServletRequest = new MockHttpServletRequest();
+
+		_mockPortletRequest = new MockLiferayResourceRequest();
+
+		_mockPortletRequest.setAttribute(
+			PortletServlet.PORTLET_SERVLET_REQUEST, _mockHttpServletRequest);
+
+		_mockPortletRequest.setAttribute(WebKeys.THEME_DISPLAY, _themeDisplay);
+
+		_mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_REQUEST, _mockPortletRequest);
+
+		_mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _themeDisplay);
+
+		_applicationsMenuInstanceConfiguration = Mockito.mock(
+			ApplicationsMenuInstanceConfiguration.class);
+
+		_panelCategoryHelper = Mockito.mock(PanelCategoryHelper.class);
+
+		_mockPortletRequest.setAttribute(
+			ApplicationListWebKeys.PANEL_CATEGORY_HELPER, _panelCategoryHelper);
+
+		PanelCategory panel1 = _newPanelCategory("panel1");
+		PanelCategory panel2 = _newPanelCategory("panel2");
+		PanelCategory panel3 = _newPanelCategory("panel3");
+
+		_panelCategories = Arrays.asList(panel1, panel2, panel3);
+	}
+
+	@Test
+	public void testShouldReturnOnlyPanelCategoriesContainsPanelApps() {
+			_assertGetPanelCategories(false, (productMenuDisplayContext) -> {
+			Mockito.when(
+				_panelCategoryHelper.getChildPanelCategories(
+					PanelCategoryKeys.ROOT, _themeDisplay)
+			).thenReturn(
+				new ArrayList()
+			);
+
+			Mockito.when(
+				_panelCategoryHelper.getChildPanelCategories(
+					PanelCategoryKeys.APPLICATIONS_MENU, _themeDisplay)
+			).thenReturn(
+				_panelCategories
+			);
+
+				List<PanelCategory> childPanelCategories = productMenuDisplayContext.getChildPanelCategories();
+
+				Assert.assertEquals(childPanelCategories.toString(), 2, childPanelCategories.size());
+
+				Mockito.verify(
+					_panelCategoryHelper
+				).getChildPanelCategories(
+					PanelCategoryKeys.ROOT, _themeDisplay
+				);
+
+				Mockito.verify(
+					_panelCategoryHelper
+				).getChildPanelCategories(
+					PanelCategoryKeys.APPLICATIONS_MENU, _themeDisplay
+				);
+
+				Mockito.verify(
+					_panelCategoryHelper, Mockito.times(3)
+				).getAllPanelApps(
+					ArgumentMatchers.matches("panel*")
+				);
```

Comment:
I think you could also put this (lines 102-122 and 137-157) inside `_assertGetPanelCategories`. What changes is just the amount of panel categories and how many times the method `getAllpanelApps` is called. For the `Mockito.never()`, I guess it may be replaced with `Mockito.times(0)`

Reply:
There are usage of both ways through Liferay code :thinking: . But I think `never` is more meaningful and readable as well, since it makes more sense to say "I'll never do that" instead of "I'll do that zero times" :laughing: